sdk/aqueduct/flow_run.py

```diff
@@ -64,6 +68,26 @@ def describe(self) -> None:
             assert param_op.spec.param is not None, "Artifact is not a parameter."
             print("* " + param_op.name + ": " + param_op.spec.param.val)
 
+    def artifact(
+        self, name: str
+    ) -> Optional[Union[TableArtifact, MetricArtifact, CheckArtifact, ParamArtifact]]:
+        """Gets the Artifact from the flow based on artifact_name"""
+        flow_run_dag = self._dag
+        artifact_from_dag = flow_run_dag.get_artifacts_by_name(name)
+
+        if artifact_from_dag is None:
+            raise ArtifactNotFoundException("The artifact name provided does not exist.")
+        elif get_artifact_type(artifact_from_dag) is ArtifactType.TABLE:
+            return TableArtifact(self._api_client, self._dag, artifact_from_dag.id, True)
+        elif get_artifact_type(artifact_from_dag) is ArtifactType.NUMBER:
+            return MetricArtifact(self._api_client, self._dag, artifact_from_dag.id, True)
+        elif get_artifact_type(artifact_from_dag) is ArtifactType.BOOL:
+            return CheckArtifact(self._api_client, self._dag, artifact_from_dag.id, True)
+        elif get_artifact_type(artifact_from_dag) is ArtifactType.PARAM:
+            return ParamArtifact(self._api_client, self._dag, artifact_from_dag.id, True)
+
+        return None
```

Comment:
Here I actually think we should raise an exception (aka fail very loudly), since this means something went wrong in our system. Is there an Internal error we can throw?

Reply:
@kenxu95 I think this error is really unlikely trigger as an artifact falls in either the four categories. However, if there will be an error, I think it is likely to be either "InvalidArtifactError"(does not exist yet in error.py) or "UnprocessableEntityError"(Exception raised for errors occured when the Aqueduct system fails to process certain inputs.)

Reply:
Or maybe we can throw the general AqueductError?

Reply:
Let's throw an `InternalAqueductError`. The reason that this needs to be internal is because its definitely something thats gone wrong in our system, and is not the user's fault.

This error should never happen, as you pointed out, but it's important in our code to make sure that we cover all assumptions. If any assumption breaks, we should also fail loudly so that the person that broke the assumption can fix the issue. In this case, the assumption is that the artifact is one of four types, but this can easily be changed (imagine someone adds a new type on the backend, but doesn't find this function and doesn't update it).